libs/market-list/src/lib/components/markets-container/markets-container.tsx

```diff
@@ -26,37 +28,53 @@ export const MarketsContainer = () => {
     [yTimestamp]
   );
 
-  const update = useCallback(({ data }: { data: MarketList_markets[] }) => {
-    if (!gridRef.current?.api) {
-      return false;
-    }
-    dataRef.current = data;
-    gridRef.current.api.refreshInfiniteCache();
-    return true;
-  }, []);
+  const update = useCallback(
+    ({ delta }: { delta: MarketDataSub_marketData }) => {
+      const update: MarketList_markets[] = [];
+      const add: MarketList_markets[] = [];
+      const remove: MarketList_markets[] = [];
+      if (!gridRef.current?.api) {
+        return false;
+      }
+      const rowNode = gridRef.current.api.getRowNode(
+        getRowId({ data: delta.market })
+      );
+      if (rowNode) {
+        const updatedData = produce<MarketList_markets>(
+          rowNode.data.data,
+          (draft: MarketList_markets) => merge(draft, delta)
+        );
+        if (updatedData !== rowNode.data.data) {
+          update.push({ ...rowNode.data, data: updatedData });
+        }
+      }
+      // @TODO - else add new market
```

Comment:
todo?

Reply:
This todo is very old. And it will need to stay with us little bit longer